.github/workflows/test.yml

```diff
@@ -0,0 +1,21 @@
+name: Test
+on: [push, pull_request, workflow_dispatch]
```

Comment:
Just curious. I saw two checks (one said push and the other said pull request) in this PR have been run, is this because we check on both `push` and `pull_request`? Will we always run the same test twice on a new commit in a PR? Is this expected?

Reply:
This is a list suggested by @lu-wang-dl . Here is the definition of events: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows

I think `push` is enough, and if we run test on every commit, `pull_request` is not necessary. I'll update the list to `[push, workflow_dispatch]`. @lu-wang-dl what do you think?

Reply:
I think the difference is on a PR from a forked repo. push event may not able to trigger the run then. You can limit the runs to master like:
```
on:
  push:
    branches:
    - master
  pull_request:
    branches:
    - master
```

Reply:
I see. The reason it triggered twice is that I was too lazy to fork the repository. In that case I think we can just have both push and pull_request, which will trigger only once from a forked repo, and twice if from a separate branch of the same repo (which wasn't too bad, I guess?).

Reply:
What do you think about this workflow: Limit the push event to master only. So when any PR is merged, it will trigger one tests. But the pushes from  other branches will not trigger any test unless you make a PR.

Reply:
Sounds good. Updated to the following. PTAL.

```
on:
  push:
    branches:
        - master
  pull_request:
  workflow_dispatch:
```